drivers/mbox/mbox_nrf_vevif_event_tx.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_vevif_event_tx
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+
+#include <hal/nrf_vpr.h>
+#include <hal/nrf_vpr_csr.h>
+#include <hal/nrf_vpr_csr_vevif.h>
+
+#if defined(CONFIG_SOC_NRF54L15_ENGA_CPUFLPR)
+#define EVENTS_IDX_MAX 17U
+#else
+#define EVENTS_IDX_MAX NRF_VPR_EVENTS_TRIGGERED_MAX
+#endif
+
+struct mbox_vevif_event_tx_conf {
+	uint32_t events_mask;
+	uint8_t events;
+};
+
+static inline bool vevif_event_tx_is_valid(const struct device *dev, uint32_t id)
+{
+	const struct mbox_vevif_event_tx_conf *config = dev->config;
+
+	return (id < EVENTS_IDX_MAX) && ((config->events_mask & BIT(id)) != 0U);
+}
+
+static int vevif_event_tx_send(const struct device *dev, uint32_t id, const struct mbox_msg *msg)
+{
+	if (!vevif_event_tx_is_valid(dev, id)) {
+		return -EINVAL;
+	}
+
+	if (msg != NULL) {
+		return -ENOTSUP;
+	}
+
+	nrf_vpr_csr_vevif_events_trigger(BIT(id));
+
+	return 0;
+}
+
+static int vevif_event_tx_mtu_get(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 0;
+}
+
+static uint32_t vevif_event_tx_max_channels_get(const struct device *dev)
+{
+	const struct mbox_vevif_event_tx_conf *config = dev->config;
+
+	return config->events;
+}
+
+static const struct mbox_driver_api vevif_event_tx_driver_api = {
+	.send = vevif_event_tx_send,
+	.mtu_get = vevif_event_tx_mtu_get,
+	.max_channels_get = vevif_event_tx_max_channels_get,
+};
+
+#define VEVIF_EVENT_REMOTE_DEFINE(inst)                                                            \
+	BUILD_ASSERT(DT_INST_PROP(inst, nordic_events) <= NRF_VPR_EVENTS_TRIGGERED_COUNT,          \
+		     "Number of events exceeds maximum");                                          \
+                                                                                                   \
+	static const struct mbox_vevif_event_tx_conf conf##inst = {                                \
+		.events = DT_INST_PROP(inst, nordic_events),                                       \
+		.events_mask = DT_INST_PROP(inst, nordic_events_mask),                             \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(inst, NULL, NULL, NULL, &conf##inst, POST_KERNEL,                    \
+			      CONFIG_MBOX_INIT_PRIORITY, &vevif_event_tx_driver_api);
+
+DT_INST_FOREACH_STATUS_OKAY(VEVIF_EVENT_REMOTE_DEFINE)
```

Comment:
is multi-instance really needed here? This is using CSR instructions... Please, check https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/mbox/mbox_nrf_vevif_local.c

Reply:
You're right, reworked to a single-instance driver.